### ISSUE TITLE
Postgres On Conflict Errors

### ISSUE THREAD

**Issue by acravenho** [poanetwork/blockscout#1129]
I believe this is what is causing the `"tcp recv: closed"` issue. 

```
2018-11-24 00:33:01 UTC:10.0.0.95(55445):poa@ether:[85100]:ERROR: canceling statement due to user request
2018-11-24 00:33:01 UTC:10.0.0.95(55445):poa@ether:[85100]:CONTEXT: while inserting index tuple (472520,99) in relation "addresses"
2018-11-24 00:33:01 UTC:10.0.0.95(55445):poa@ether:[85100]:STATEMENT: INSERT INTO "addresses" AS a0 ("contract_code","fetched_coin_balance","fetched_coin_balance_block_number","hash","inserted_at","updated_at") VALUES (DEFAULT,$1,$2,$3,$4,$5),(DEFAULT,$6,$7,$8,$9,$10),(DEFAULT,$11,$12,$13,$14,$15),(DEFAULT,$16,$17,$18,$19,$20),(DEFAULT,$21,$22,$23,$24,$25),(DEFAULT,$26,$27,$28,$29,$30),(DEFAULT,$31,$32,$33,$34,$35),(DEFAULT,$36,$37,$38,$39,$40),(DEFAULT,$41,$42,$43,$44,$45),(DEFAULT,$46,$47,$48,$49,$50),($51,$52,$53,$54,$55,$56),(DEFAULT,$57,$58,$59,$60,$61),(DEFAULT,$62,$63,$64,$65,$66),(DEFAULT,$67,$68,$69,$70,$71),(DEFAULT,$72,$73,$74,$75,$76),(DEFAULT,$77,$78,$79,$80,$81),(DEFAULT,$82,$83,$84,$85,$86),(DEFAULT,$87,$88,$89,$90,$91),(DEFAULT,$92,$93,$94,$95,$96),($97,$98,$99,$100,$101,$102),($103,$104,$105,$106,$107,$108),(DEFAULT,$109,$110,$111,$112,$113),(DEFAULT,$114,$115,$116,$117,$118),(DEFAULT,$119,$120,$121,$122,$123),(DEFAULT,$124,$125,$126,$127,$128),(DEFAULT,$129,$130,$131,$132,$133),(DEFAULT,$134,$135,$136,$137,$138),(DEFAULT,$139,$140,$141,$142,$143),(DEFAULT,$144,$145,$146,$147,$148),(DEFAULT,$149,$150,$151,$152,$153),(DEFAULT,$154,$155,$156,$157,$158),(DEFAULT,$159,$160,$161,$162,$163),(DEFAULT,$164,$165,$166,$167,$168),(DEFAULT,$169,$170,$171,$172,$173),(DEFAULT,$174,$175,$176,$177,$178),(DEFAULT,$179,$180,$181,$182,$183),(DEFAULT,$184,$185,$186,$187,$188),(DEFAULT,$189,$190,$191,$192,$193),(DEFAULT,$194,$195,$196,$197,$198),(DEFAULT,$199,$200,$201,$202,$203),(DEFAULT,$204,$205,$206,$207,$208),(DEFAULT,$209,$210,$211,$212,$213),(DEFAULT,$214,$215,$216,$217,$218),(DEFAULT,$219,$220,$221,$222,$223),(DEFAULT,$224,$225,$226,$227,$228),(DEFAULT,$229,$230,$231,$232,$233),(DEFAULT,$234,$235,$236,$237,$238),(DEFAULT,$239,$240,$241,$242,$243),(DEFAULT,$244,$245,$246,$247,$248),(DEFAULT,$249,$250,$251,$252,$253),(DEFAULT,$254,$255,$256,$257,$258),(DEFAULT,$259,$260,$261,$262,$263),(DEFAULT,$264,$265,$266,$267,$268),(DEFAULT,$269,$270,$271,$272,$273),(DEFAULT,$274,$275,$276,$277,$278),(DEFAULT,$279,$280,$281,$282,$283),(DEFAULT,$284,$285,$286,$287,$288),(DEFAULT,$289,$290,$291,$292,$293),(DEFAULT,$294,$295,$296,$297,$298),(DEFAULT,$299,$300,$301,$302,$303),(DEFAULT,$304,$305,$306,$307,$308),(DEFAULT,$309,$310,$311,$312,$313),(DEFAULT,$314,$315,$316,$317,$318),(DEFAULT,$319,$320,$321,$322,$323),(DEFAULT,$324,$325,$326,$327,$328),(DEFAULT,$329,$330,$331,$332,$333),(DEFAULT,$334,$335,$336,$337,$338),($339,$340,$341,$342,$343,$344),(DEFAULT,$345,$346,$347,$348,$349),(DEFAULT,$350,$351,$352,$353,$354),(DEFAULT,$355,$356,$357,$358,$359),(DEFAULT,$360,$361,$362,$363,$364),(DEFAULT,$365,$366,$367,$368,$369),(DEFAULT,$370,$371,$372,$373,$374),(DEFAULT,$375,$376,$377,$378,$379),(DEFAULT,$380,$381,$382,$383,$384),(DEFAULT,$385,$386,$387,$388,$389),(DEFAULT,$390,$391,$392,$393,$394),(DEFAULT,$395,$396,$397,$398,$399),(DEFAULT,$400,$401,$402,$403,$404),(DEFAULT,$405,$406,$407,$408,$409),(DEFAULT,$410,$411,$412,$413,$414),(DEFAULT,$415,$416,$417,$418,$419),(DEFAULT,$420,$421,$422,$423,$424),(DEFAULT,$425,$426,$427,$428,$429),(DEFAULT,$430,$431,$432,$433,$434),(DEFAULT,$435,$436,$437,$438,$439),(DEFAULT,$440,$441,$442,$443,$444),(DEFAULT,$445,$446,$447,$448,$449),(DEFAULT,$450,$451,$452,$453,$454),(DEFAULT,$455,$456,$457,$458,$459),(DEFAULT,$460,$461,$462,$463,$464),(DEFAULT,$465,$466,$467,$468,$469),(DEFAULT,$470,$471,$472,$473,$474),(DEFAULT,$475,$476,$477,$478,$479),(DEFAULT,$480,$481,$482,$483,$484),(DEFAULT,$485,$486,$487,$488,$489),(DEFAULT,$490,$491,$492,$493,$494),(DEFAULT,$495,$496,$497,$498,$499),(DEFAULT,$500,$501,$502,$503,$504),(DEFAULT,$505,$506,$507,$508,$509),(DEFAULT,$510,$511,$512,$513,$514),(DEFAULT,$515,$516,$517,$518,$519),(DEFAULT,$520,$521,$522,$523,$524),(DEFAULT,$525,$526,$527,$528,$529),(DEFAULT,$530,$531,$532,$533,$534) ON CONFLICT ("hash") DO UPDATE SET "contract_code" = COALESCE(a0."contract_code", EXCLUDED.contract_code), "fetched_coin_balance" = CASE WHEN EXCLUDED.fetched_coin_balance_block_number IS NOT NULL AND
(a0."fetched_coin_balance_block_number" IS NULL OR
EXCLUDED.fetched_coin_balance_block_number >= a0."fetched_coin_balance_block_number") THEN
EXCLUDED.fetched_coin_balance
ELSE a0."fetched_coin_balance"
END
, "fetched_coin_balance_block_number" = CASE WHEN EXCLUDED.fetched_coin_balance_block_number IS NOT NULL AND
(a0."fetched_coin_balance_block_number" IS NULL OR
EXCLUDED.fetched_coin_balance_block_number >= a0."fetched_coin_balance_block_number") THEN
EXCLUDED.fetched_coin_balance_block_number
ELSE a0."fetched_coin_balance_block_number"
END
RETURNING "updated_at", "inserted_at", "contract_code", "fetched_coin_balance_block_number", "fetched_coin_balance", "hash"
```

### Environment

* Operating System: Linux
* Chain: Mainnet

### Steps to reproduce

1. Index a sufficient amount of Mainnet.
2. Very large batches of coin balances are being inserted into the DB. 
3. I'm seeing this for logs, internal transactions, and addresses.

**Comment by KronicDeth** [comment]
`Explorer.Repo.safe_insert_all` already chunks the insert to 1000 rows, so this not working must mean that `1000` rows is too many.

**Comment by KronicDeth** [comment]
Resolved by the various ports of #1185 and other associated changes.
